### PR TITLE
docs(Admin): document i18nprovider and store for AdminContext

### LINKED ISSUE #7520: v4: custom <Admin /> removes all labels (i18n?)
When using the AdminContext instead of `<Admin />` all labels change into what I assume are keys.

Upon checking the code [I think this is the issue](https://github.com/marmelab/react-admin/blob/v4.0.0-rc.1/packages/react-admin/src/Admin.tsx#L146,L149):

```js
Admin.defaultProps = {
    i18nProvider: defaultI18nProvider,
    store: localStorageStore(),
};
```

These aren't mentioned in the docs. I think this should be added to the docs, or should be moved into `<AdminContext />`.

**What you were expecting:**

I was expecting nothing to change and everything to look the same.

**What happened instead:**

After following this guide: https://marmelab.com/react-admin/doc/4.0/Admin.html#unplugging-the-admin-using-admincontext-and-adminui

I got the following result:

image

**Related code:**

My <App /> looks like this now (removed most resources and the irrelevant bits for brevity):

```tsx
function App() {
  return (
    <AdminContext authProvider={authProvider} dataProvider={withUpload}>
      <AsyncResources />
    </AdminContext>
  );
}

function AsyncResources() {
  return (
    <AdminUI layout={MyLayout} loginPage={SignIn} disableTelemetry={true}>
      <Resource name="profiles" list={ProfileList} show={ProfileShow} edit={ProfileEdit} create={ProfileCreate} />
      <Resource name="users" list={UserList} show={UserShow} edit={UserEdit} create={UserCreate} />
    </AdminUI>
  );
}
```

**Environment**

* React-admin version: 4.0.0-rc.1
* React version: 17.0.2
* Browser: Chrome 
* Stack trace (in case of a JS error): N/A

### PR DESCRIPTION
**Pull request by RWOverdijk**
Update docs & docblock.

fixes #7520

### CODE REVIEW

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Comment by RWOverdijk** [comment]
@fzaninotto I did find that changing auth status doesn't reload the resources. The render gets called but navigation doesn't change. Mentioning this here before going through the issue process.

For example, before logging in all my resources return null (no permission, because I don't know who you are). After logging in I get the [`ready` component](https://marmelab.com/react-admin/doc/4.0/Admin.html#ready) because resources don't get registered.

Code helps, so here's my implementation:

```ts
function App() {
  return (
    <AdminContext
      authProvider={authProvider}
      dataProvider={withUpload}
      i18nProvider={defaultI18nProvider}
      store={store}
    >
      <GuardedResources />
    </AdminContext>
  );
}

function GuardedResources() {
  const { isLoading, guard } = usePermissionGuard();
  const composedResources = useMemo(() => {
    if (isLoading) return null;

    return resources.map(({ permissionResource: r, components, ...props }) => {
      if (!guard.isAllowed(r)) return null;

      return <Resource key={`r_${r}`} {...guard.filterComponents(r, components)} {...props} />;
    });
  }, [isLoading, guard]);
  
  return (
    <AdminUI layout={MyLayout} loginPage={SignIn} disableTelemetry={true} requireAuth>
      {composedResources}
    </AdminUI>
  );
}

export default App;
```

`isAllowed` is linked to the custom permission system which is simply a map of resources and actions. The `composedResources` memo gets called but the UI doesn't update and `ready` still shows up.

------

TL;DR; the resources don't get reloaded when logging in causing me to see the Admin.ready component until I refresh the page. Is this a bug or just shitty implementation on my side?

**Comment by fzaninotto** [comment]
@RWOverdijk I'm not sure I understand the problem. As far as I know, the current implementation of dynamic resources based on permissions works (it's backed by unit tests). If you can provide a repro based on the simple example showing that it doesn't work, please open a new issue about it.

**Comment by RWOverdijk** [comment]
Will do.
